mmpy_bot/function.py

```diff
@@ -115,7 +112,7 @@ def __init__(
         if _function is not None:
             self.name = _function.__qualname__
 
-        argspec = list(inspect.signature(_function).parameters.keys())
+        argspec = list(islice(inspect.signature(_function).parameters.keys(), 2))
         if not argspec[:2] == ["self", "message"]:
```

Comment:
Given the `islice` in the previous line, `[:2]` is redundant.

```suggestion
        if not argspec == ["self", "message"]:
```

Reply:
I don't expect functions to have so many arguments that this would become a significant performance bottleneck. It's fair to say that it is a micro optimization.

There are now also tests failing due to the `islice` change so please include the test correction in this PR or revert the `islice` line.